### PR TITLE
Element-R: Implement `VerificationRequest.accept`

### PR DESCRIPTION
**Pull request by richvdh**
Part of https://github.com/vector-im/element-web/issues/25320

~~Based on https://github.com/matrix-org/matrix-js-sdk/pull/3528~~

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

**Comment by richvdh** [comment]
needs https://github.com/matrix-org/matrix-rust-sdk/pull/2177 to pass the tests :-S

File: src/rust-crypto/verification.ts [hunk 1]
```diff
@@ -198,8 +214,28 @@ export class RustVerificationRequest
      *
      * @returns Promise which resolves when the event has been sent.
      */
-    public accept(): Promise<void> {
-        throw new Error("not implemented");
+    public async accept(): Promise<void> {
+        if (this.inner.phase() != RustSdkCryptoJs.VerificationRequestPhase.Requested || this._accepting) {
```
Comment (alunturner) [inline src/rust-crypto/verification.ts:218]: Just asking in case this is a typo - is this deliberately using `!=`?

**Review by alunturner — APPROVED** [review]

File: src/rust-crypto/verification.ts (hunk 1 above)
Comment (richvdh) [reply src/rust-crypto/verification.ts:218]: nope, it's a typo

Comment (richvdh) [reply src/rust-crypto/verification.ts:218]:
```suggestion
        if (this.inner.phase() !== RustSdkCryptoJs.VerificationRequestPhase.Requested || this._accepting) {
```

File: yarn.lock [hunk 2]
```diff
@@ -1427,9 +1427,9 @@
     lodash "^4.17.21"
 
 "@matrix-org/matrix-sdk-crypto-js@^0.1.0-alpha.11":
-  version "0.1.0-alpha.11"
-  resolved "https://registry.yarnpkg.com/@matrix-org/matrix-sdk-crypto-js/-/matrix-sdk-crypto-js-0.1.0-alpha.11.tgz#24d705318c3159ef7dbe43bca464ac2bdd11e45d"
-  integrity sha512-HD3rskPkqrUUSaKzGLg97k/bN+OZrkcX7ODB/pNBs/jqq+/A0wDKqsszJotzFwsQcDPpWn78BmMyvBo4tLxKjw==
+  version "0.1.0"
+  resolved "https://registry.yarnpkg.com/@matrix-org/matrix-sdk-crypto-js/-/matrix-sdk-crypto-js-0.1.0.tgz#766580036d4df12120ded223e13b5640e77db136"
+  integrity sha512-ra/bcFdleC1iRNms2I96UXA0NvQYWpMsHrV5EfJRS7qV1PtnQNvgsvMfjMbkx8QT2ErEmIhsvB5fPCpfp8BSuw==
```
Comment (richvdh) [inline yarn.lock:1432]: 0.1.0 fixes a bug in cancellation which you'll likely never notice unless you happen to be an integration test

**Review by t3chguy — APPROVED** [review]

**Pull request merged**